CONTRIBUTING.adoc

```diff
@@ -2,45 +2,135 @@
 :toc: left
 :reproducible:
 
-== Setting up and working with our development tools
```

Comment:
This file is best reviewed rendered -> https://github.com/input-output-hk/plutus/blob/0323c8bb2919b0b7b63d9601586177380fd37bec/CONTRIBUTING.adoc

Reply:
The section `How to setup you development environment` has been largely re-written.
All other sections are mostly unchanged (except for the occasional amend `nix-shell` -> `nix develop`)